src/client/method-unit-actions.c

```diff
@@ -0,0 +1,388 @@
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        BC_INTERFACE_BASE_NAME,
+                        client->object_path,
+                        NODE_INTERFACE,
+                        "ThawUnit",
+                        &error,
+                        &result,
+                        "s",
+                        unit);
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        printf("Unit %s thaw operation done\n", unit);
+
+        return 0;
+}
+
+int method_start(Command *command, void *userdata) {
+        return method_lifecycle_action_on(userdata, command->opargv[0], command->opargv[1], "StartUnit");
+}
+
+int method_stop(Command *command, void *userdata) {
+        return method_lifecycle_action_on(userdata, command->opargv[0], command->opargv[1], "StopUnit");
+}
+
+int method_restart(Command *command, void *userdata) {
+        return method_lifecycle_action_on(userdata, command->opargv[0], command->opargv[1], "RestartUnit");
+}
+
+int method_reload(Command *command, void *userdata) {
+        return method_lifecycle_action_on(userdata, command->opargv[0], command->opargv[1], "ReloadUnit");
+}
+
+int method_freeze(Command *command, void *userdata) {
+        return method_freeze_unit_on(userdata, command->opargv[0], command->opargv[1]);
+}
+
+int method_thaw(Command *command, void *userdata) {
+        return method_thaw_unit_on(userdata, command->opargv[0], command->opargv[1]);
+}
+
+int method_enable(Command *command, void *userdata) {
+        int r = 0;
+        r = method_enable_unit_on(userdata, command->opargv[0], &command->opargv[1], command->opargc - 1, 0, 0);
+        if (r < 0) {
+                fprintf(stderr,
+                        "Failed to enable the units on node [%s] - %s",
+                        command->opargv[0],
+                        strerror(-r));
+        } else {
+                r = method_daemon_reload_on(userdata, command->opargv[0]);
+        }
+
+        return r;
+}
+
+int method_disable(Command *command, void *userdata) {
+        int r = 0;
+        r = method_disable_unit_on(userdata, command->opargv[0], &command->opargv[1], command->opargc - 1, 0);
+        if (r < 0) {
+                fprintf(stderr,
+                        "Failed to disable the units on node [%s] - %s",
+                        command->opargv[0],
+                        strerror(-r));
+        } else {
+                r = method_daemon_reload_on(userdata, command->opargv[0]);
+        }
+
+        return r;
+}
+
+int method_daemon_reload(Command *command, void *userdata) {
+        return method_daemon_reload_on(userdata, command->opargv[0]);
+        ;
```

Comment:
I guess this line should be removed, right?

Reply:
Oops :)